lib/scholar/shared.ex

```diff
@@ -66,4 +66,27 @@ defmodule Scholar.Shared do
               "invalid value for :weights option: expected list or tensor of positive numbers of size #{num_samples}, got: #{inspect(weights)}"
     end
   end
+
+  deftransform valid_broadcast?(n_dims, shape1, shape2) do
+    if tuple_size(shape1) != tuple_size(shape2) do
+      raise ArgumentError,
+            "expected shapes to have same rank, got #{inspect(tuple_size(shape1))} and #{inspect(tuple_size(shape2))}"
+    end
+
+    valid_broadcast(Enum.to_list(0..(n_dims - 1)), shape1, shape2)
```

Comment:
Instead of creating a list, start with `n_dims` and decrement one while `n_dims >= 0`.

Reply:
Yeah, now I'm wondering what I was thinking when I write this function using list 😅